java-test-samples/async-lambda-dynamodb/src/test/java/com/amazon/aws/sample/TestAsyncTransformation.java

```diff
@@ -0,0 +1,216 @@
+public class TestAsyncTransformation {
+
+    private static S3Client s3Client;
+
+    private static DynamoDbClient dynamoDbClient;
+
+    private static String sourceBucketName = null;
+
+    private static String destinationBucketName = null;
+
+    private static String recordTransformationTable = null;
+
+    private static final String MESSAGE = "this message was created during an integration test";
+    private static final String MESSAGE_TO_UPPER_CASE = MESSAGE.toUpperCase();
+
+    @BeforeAll
+    public static void runBeforeAll() {
+        s3Client = S3Client.builder()
+                .httpClient(ApacheHttpClient.create())
+                .build();
+
+        dynamoDbClient = DynamoDbClient.builder().build();
+
+        String awsSamStackName = System.getenv("AWS_SAM_STACK_NAME");
+
+        DescribeStacksRequest describeStacksRequest = DescribeStacksRequest.builder().stackName(awsSamStackName)
+                .build();
+
+        CloudFormationClient cloudFormationClient = CloudFormationClient.builder().build();
+
+        DescribeStacksResponse describeStacksResponse = cloudFormationClient.describeStacks(describeStacksRequest);
+        assertNotNull(describeStacksResponse, "describeStacksResponse may not be null");
+        assertTrue(describeStacksResponse.hasStacks(), "describeStacksResponse may not be empty");
+
+        for (Stack stack : describeStacksResponse.stacks()) {
+            for (Output output : stack.outputs()) {
+                if ("SourceBucketName".equals(output.outputKey())) {
+                    sourceBucketName = output.outputValue();
+                } else if ("DestinationBucketName".equals(output.outputKey())) {
+                    destinationBucketName = output.outputValue();
+                } else if ("RecordTransformationTable".equals(output.outputKey())) {
+                    recordTransformationTable = output.outputValue();
+                }
+            }
+        }
+        assertNotNull(sourceBucketName, "sourceBucketName may not be null");
+        assertNotNull(destinationBucketName, "destinationBucketName may not be null");
+    }
+
+    @Test
+    public void testAsyncTransformation_when_txt() {
+        String key = String.format("%s.txt", UUID.randomUUID().toString());
+
+        PutObjectRequest putObjectRequest = PutObjectRequest.builder().bucket(sourceBucketName)
+                .key(key).build();
+
+        PutObjectResponse putObjectResponse = s3Client.putObject(putObjectRequest, RequestBody.fromString(MESSAGE));
+        assertNotNull(putObjectResponse, "putObjectResponse may not be null");
+
+        if (recordTransformationTable != null) {
+            try {
+                Callable<String> callable = getContentFromDynamoDB(recordTransformationTable, key);
+
+                String contentFromRecordTransformationTable = callable.call();
+
+                assertEquals(contentFromRecordTransformationTable, contentFromRecordTransformationTable,
```

Comment:
Not sure if I follow this line. Expected and Actual values are the same here in assertEquals?